### PR TITLE
Added fix for select all and exports

### PR DESCRIPTION
**Pull request by abist**
Added fixes for all combinations of Selection and Save as Results. Tested the following scenarios -
- [x] Ctrl + A -> right click -> Save as
- [x] Ctrl + A -> Save as button
- [x] Right click -> Select All -> right click -> Save as
- [x] Right click -> Select All -> Save as button
- [x] Mouse drag selection -> Right Click -> Save as
- [x] No selection -> Save as button

Accompanying PR in slickgrid here - https://github.com/anthonydresser/SlickGrid/pull/24

Fixes -
https://github.com/Microsoft/sqlopsstudio/issues/2372
https://github.com/Microsoft/sqlopsstudio/issues/931
https://github.com/Microsoft/sqlopsstudio/issues/2482

### CODE REVIEW

File: src/sql/parts/query/editor/actions.ts [hunk 1]
```diff
@@ -42,6 +42,14 @@ function mapForNumberColumn(ranges: Slick.Range[]): Slick.Range[] {
 	}
 }
 
+function selectedRowsToRange(selectedRows: number[], columns: number): Slick.Range[] {
+	if (selectedRows) {
```
Comment (kburtram) [inline src/sql/parts/query/editor/actions.ts]: you don't need to also check 'selectedRows.length > 0'?

**Comment by anthonydresser** [comment]
If this actually solves the problem, I think there is something else going on, since the selection for the context is generated when the context menu is created, and it gets the selection from the selection model directly.

**Comment by anthonydresser** [comment]
If this is because the icon don't have the correct context, I've known of this issue and the right solution is to change the selection in the context for the actionbar, because there are other problems that this doesn't solve in those scenarios.

**Comment by abist** [comment]
@anthonydresser The issue was that the context didn't have any selection, though it was correct. Would changing the context in the actionBar and related changes be a big task? If yes, then perhaps we can keep that task as backlog since we want these fixes quick.

**Comment by anthonydresser** [comment]
@abist it shouldn't, can you check if bug/selectionOnGridContext fixes the same problems you are trying to fix?

**Comment by abist** [comment]
@anthonydresser I'm going to go ahead and merge this fix. We can open a separate issue if there's a better solution that fixes all the scenarios later.

**Comment by anthonydresser** [comment]
`bug/selectionOnGridContext` now fixes all scenarios without needing change to the slickgrid library. It also fixes other scenarios.

**Comment by anthonydresser** [comment]
Actions generally shouldn't modify their contexts because the contexts are there to tell them what to do, so modifying the context doesn't really make sense. I also don't think the slickgrid changes are necessary since we have access to everything that slickgrid does.

**Comment by abist** [comment]
Since you've worked more on this area, it's probably a better idea to take your fix since you're more familiar. We just need to be sure every scenario works as expected :)

**Comment by abist** [comment]
Fixed in another PR.

**Pull request closed without merging**